distributed/protocol/utils.py

```diff
@@ -67,6 +67,9 @@ def merge_frames(header, frames):
     if all(len(f) == l for f, l in zip(frames, lengths)):
         return frames
 
+    if len(lengths) == 1:
+        return [b"".join(map(ensure_bytes, frames))]
```

Comment:
Does it not imply only one frame, so you can `return ensure_bytes(frames[0])` ?

Reply:
No it doesn't. It implies there was only one frame. However we may already have split it up during serialization. Noted the line where this happens below.

Reply:
Is it worth, then, checking for exactly one frame, to avoid the copy during `join`?

Reply:
Not sure. Was trying to integrate your feedback above. Happy to revert it if you prefer 🙂

Reply:
Tried to respond inline, but GitHub didn't like it. Moved here: https://github.com/dask/distributed/pull/3639#pullrequestreview-424799455